app/containers/TuneContainer/saga.js

```diff
@@ -12,10 +15,35 @@ export function* getItuneSongs(action) {
   if (ok) {
     yield put(successGetItuneSongs(data));
   } else {
-    yield put(failureGetItuneSongs(data));
+    const error = data?.originalError?.message ?? translate('something_went_wrong');
+    yield put(failureGetItuneSongs(error));
   }
 }
 
+export function* getTrackDetails(action) {
+  let songsData;
+  if (action.testData) {
+    songsData = action.testData;
+  } else {
+    songsData = yield select(selectSongsData());
+  }
+
+  const songItem = songsData?.results?.find((_song) => _song.trackId?.toString() === action.songId);
```

Comment:
generally `_variableNames` are used to denote unused variable and some lodash operations. Don't use it here. `song` makes much more semantic sense.